anvill/python/anvill/binja/bnprogram.py

```diff
@@ -78,6 +82,114 @@ def get_variable_impl(self, address):
+                and var_tinfo.element_type.type_class == bn.TypeClass.FunctionTypeClass
+            ):
+                return True
+            return False
+
+        if bn_var is None:
+            return None
+
+        if not _is_function_pointer(bn_var):
+            return None
+
+        arch = self._arch
+        func_tinfo = bn_var.type.element_type
+
+        # TODO(akshayk): The type information may not have calling convention
+        # information. It does not recover the parameters and get the function
+        # in such case. A working solution could be handing the default calling
+        # convention for each architecture. This is in todo list.
+        if func_tinfo.calling_convention is None:
+            return
+
+        calling_conv = CallingConvention(arch, func_tinfo)
+
+        func_type = self.type_cache.get(func_tinfo)
+
+        index = 0
+        param_list = []
+        stack_offset = arch.stack_offset()
+        for var in func_tinfo.parameters:
+            var_type = var.type
+            reg_name = None
+
+            if var_type is None:
+                continue
+
+            arg_type = self.type_cache.get(var_type)
+
+            if (
+                bn.TypeClass.IntegerTypeClass == var_type.type_class
+                or bn.TypeClass.PointerTypeClass == var_type.type_class
+            ):
+                reg_name = calling_conv.next_int_arg_reg
+
+            elif bn.TypeClass.FloatTypeClass == var_type.type_class:
+                reg_name = calling_conv.next_float_arg_reg
+
+            elif bn.TypeClass.NamedTypeReferenceClass == var_type.type_class:
+                # The function paramater could be named alias of a float type.
+                if isinstance(arg_type, FloatingPointType):
+                    reg_name = calling_conv.next_float_arg_reg
+                else:
+                    reg_name = calling_conv.next_int_arg_reg
+
+            elif bn.TypeClass.VoidTypeClass == var_type.type_class:
+                reg_name = "invalid void"
+
+            else:
+                reg_name = None
+                raise AnvillException(
+                    "No variable type defined for function parameters"
+                )
+
+            if reg_name:
+                loc = Location()
+                loc.set_register(reg_name.upper())
+                loc.set_type(arg_type)
+                param_list.append(loc)
+
+            else:
+                loc = Location()
+                loc.set_memory(self._bv.arch.stack_pointer.upper(), stack_offset)
+                loc.set_type(arg_type)
+                param_list.append(loc)
+                stack_offset = stack_offset + arch.pointer_size()
+
+            index += 1
+
+        ret_list = []
+        retTy = self.type_cache.get(func_tinfo.return_value)
+        if not isinstance(retTy, VoidType):
+            for reg in calling_conv.return_regs:
+                loc = Location()
+                loc.set_register(reg.upper())
```

Comment:
Ditto here

Reply:
I don't quite understand the logic here. If the return type is an integer, then will it appear in `rax`, `rdx`, `rcx`, `st0`, `st1`, etc. for amd64?